operator/v1/types_dns.go

```diff
@@ -101,6 +100,39 @@ type DNSSpec struct {
 	// +optional
 	// +kubebuilder:default=Normal
 	LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+
+	// cache describes the caching configuration that applies to all server blocks listed in the Corefile.
+	// This field allows a cluster admin to optionally configure:
+	// * successTTL which is a duration for which positive responses should be cached.
+	// * denialTTL which is a duration for which negative responses should be cached.
+	// If this is not configured, OpenShift will configure success and denial caching with a default value that is
+	// subject to change. At the time of writing, the default successTTL is 900 seconds and the default denialTTL is 30
+	// seconds or as noted in the respective Corefile for your version of OpenShift.
+	// +optional
+	Cache DNSCache `json:"cache,omitempty"`
+}
+
+// DNSCache defines the fields for configuring DNS caching.
+type DNSCache struct {
+	// successTTL is optional and specifies the amount of time that a positive response should be cached.
+	//
+	// If configured, it must be a value of 1s (1 second) or greater up to a theoretical maximum of several years.
+	// If not configured, a default value of 900 seconds will be used unless noted otherwise in the respective
+	// Corefile for your version of OpenShift.
+	// +kubebuilder:validation:Pattern=^(([1-9][0-9]*(\.[0-9]+)?(s|m|h|d))+)$
+	// +kubebuilder:validation:Type:=string
+	// +optional
+	SuccessTTL *metav1.Duration `json:"successTTL,omitempty"`
+
+	// denialTTL is optional and specifies the amount of time that a negative response should be cached.
+	//
+	// If configured, it must be a value of 1s (1 second) or greater up to a theoretical maximum of several years.
+	// If this is not configured, a default value of 30s (30 seconds) will be used unless noted otherwise in the
+	// respective Corefile for your version of OpenShift.
+	// +kubebuilder:validation:Pattern=^(([1-9][0-9]*(\.[0-9]+)?(s|m|h|d))+)$
+	// +kubebuilder:validation:Type:=string
+	// +optional
+	DenialTTL *metav1.Duration `json:"denialTTL,omitempty"`
```

Comment:
Is a zero value valid here? What would happen if I put this value as `0s`?

Looking at the validation, that doesn't appear to be a valid option, in which case, I think this doesn't need to be a pointer, let's make it a reference instead and your controller can default it when it sees a 0 value

Reply:
Fixed in the latest push!